packages/tanstack/src/query-provider/index.tsx

```diff
@@ -0,0 +1,30 @@
+'use client';
+
+import { type JSX, type PropsWithChildren } from 'react';
+import { type QueryClient, QueryClientProvider } from '@tanstack/react-query';
+
+/**
+ * TanStack Query Provider component.
+ * @param root0 - The root object.
+ * @param root0.children - The child components.
+ * @param root0.client - The TanStack Client instance.
```

Comment:
Conventions in the rest of the codebase has us calling these `props.` not `root#`